lib/src/builder/codegen/accessors_generator.dart

```diff
@@ -65,15 +65,22 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
     final typeParamsOnClass = typeParameters?.toSource() ?? '';
     final typeParamsOnSuper = removeBoundsFromTypeParameters(typeParameters);
 
-    final deprecationLine = internalGeneratedMemberDeprecationLine(
+    final generatedCodeUseOnlyDeprecation = internalGeneratedMemberDeprecationLine(
         additionalMessageStringLiteral: "' EXCEPTION: this may be used in legacy boilerplate until'"
             "' it is transitioned to the new mixin-based boilerplate.'");
 
     StringBuffer generatedClass = StringBuffer();
+    // Preserve annotations authored by the consumer on any public APIs generated by the builder
+    for (final annotation in node.metadata) {
+      if (!accessorsMixinName.startsWith('_') &&
```

Comment:
Why only public APIs? I think we should do this regardless of the visibility of the class; if a consumer deprecates a private mixin and uses the generated code, they should get a warning as well

Reply:
hmmm... good point

Reply:
Addressed in https://github.com/Workiva/over_react/pull/623/commits/3d22d809afb759376e3543a1fd32721a1159d66c